File: LaravelAndPhonePackages/Webinar_networking_interview_and_Podcast_Laravel_package/src/Models/NetworkingParticipant.php
<?php

namespace Jobi\WebinarNetworkingInterviewPodcast\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class NetworkingParticipant extends Model
{
    use HasFactory;

    protected $fillable = [
        'networking_session_id',
        'user_id',
        'current_partner_id',
        'rotation_position',
        'joined_at',
        'status',
    ];

    protected $casts = [
        'joined_at' => 'datetime',
    ];

    public function session(): BelongsTo
    {
        return $this->belongsTo(NetworkingSession::class, 'networking_session_id');
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(config('auth.providers.users.model'));
    }

    public function currentPartner(): BelongsTo
    {
        return $this->belongsTo(config('auth.providers.users.model'), 'current_partner_id');
    }
}

File: LaravelAndPhonePackages/Webinar_networking_interview_and_Podcast_Flutter_addon/lib/src/models/networking_participant.dart
import 'model_utils.dart';

class NetworkingParticipant {
  final int id;
  final int networkingSessionId;
  final int userId;
  final int? currentPartnerId;
  final int rotationPosition;
  final DateTime? joinedAt;
  final String status;

  NetworkingParticipant({
    required this.id,
    required this.networkingSessionId,
    required this.userId,
    required this.rotationPosition,
    required this.status,
    this.currentPartnerId,
    this.joinedAt,
  });

  factory NetworkingParticipant.fromJson(Map<String, dynamic> json) {
    return NetworkingParticipant(
      id: json['id'] as int,
      networkingSessionId: json['networking_session_id'] as int,
      userId: json['user_id'] as int,
      currentPartnerId: json['current_partner_id'] as int?,
      rotationPosition: json['rotation_position'] as int? ?? 1,
      joinedAt: parseDateTime(json['joined_at']),
      status: json['status'] as String? ?? 'registered',
    );
  }
}
